CSharpSourceCode/Battle/StatusEffects/StatusEffectMissionLogic.cs

```diff
@@ -15,28 +16,59 @@ namespace TOW_Core.Battle.StatusEffects
 {
     public class StatusEffectMissionLogic : MissionLogic
     {
+        private Dictionary<int, Agent> _activeAgents;
+        private bool init;
+
+      
         public override void OnAgentCreated(Agent agent)
         {
+            if (_activeAgents == null&&!init)
+            {
+                _activeAgents = new Dictionary<int, Agent>(); 
+                init = true;
+            }
+            
             if (agent.IsHuman)
             {
                 StatusEffectComponent effectComponent = new StatusEffectComponent(agent);
                 agent.AddComponent(effectComponent);
+                _activeAgents?.Add(agent.Index, agent);
```

Comment:
_activeAgents is checked twice (null ref)

Reply:
changed this thanks